Mathlib/Tactic/Change.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2023 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import Mathlib.Tactic.Basic
+/-!
+
+# Tactic `change? _`
+
+This tactic is used to replace the goal with a definitionally equal one.
+The `_` is a general term and is intended to be used to guide which unfolding you want
+Lean to perform.
+-/
+
+/--  `change? term?` unifies the optional `term?` with the current goal, defaulting to the
+goal, if `term?` is not present.
+It then prints in the infoView the unified term.
+This is useful to replace the main goal after a `dsimp`.
+```lean
+example : (fun x : Nat => x) 0 = 1 := by
+  change? 0 = _  -- `change 0 = 1`
+```
+-/
```

Comment:
```suggestion
/--`change? term` unifies `term` with the current goal, then suggests explicit `change` syntax that 
uses the resulting unified term.

If `term` is not present, `change?` suggests the current goal itself. This is useful after tactics 
which transform the goal while maintaining definitional equality, such as `dsimp`; those preceding 
tactic calls can then be deleted.
```lean
example : (fun x : Nat => x) 0 = 1 := by
  change? 0 = _  -- `Try this: change 0 = 1`
```
-/
```